src/circuit/circuit_frame.rs

```diff
@@ -3030,9 +3166,9 @@ mod tests {
             assert!(delta == Delta::Equal);
 
             //println!("{}", print_cs(&cs));
-            assert_eq!(31211, cs.num_constraints());
+            assert_eq!(34530, cs.num_constraints());
```

Comment:
ouch

Reply:
This seems really expensive. At least some of this (the continuations) should be reduced when we get around to optimizing/refactoring `reduce_cons`, I think. Hopefully some of the consing can also be shared in different paths. 

That said, even though I like BEGIN1, I am starting to think it can't earn its place in the circuit. One can always get the effect if needed with an explicit LET, and as @namin points out, we can add a macro too. Maybe we should take this as a sign to make sure we deal with the macro hygiene in a satisfactory way (which I think need not be so bad and as is a worthwhile thing to think through anyway).

This may be a useful pressure to push whatever can be done that way into libraries/macros. When we can afford more circuit (because of coprocessors) this may be less relevant, but for now it may be more important to preserve a minimal core and spend our constraint budget wisely.

Let's think about this a little more. @emmorais Can you try to understand what the likely cost of BEGIN1 (above and beyond BEGIN) is:
1. in the current implementation.
2. hypothetically, after this has all been better optimized.

That information will help inform a better decision about what to do here. I think this has been a good exercise forcing us to think about tradeoffs, whatever we decide.

Reply:
Yes. Ultimately this is manual macroexpansion of something that doesn't earn its place in the circuit. Thanks for your patience.